decidim-core/app/controllers/decidim/links_controller.rb

```diff
@@ -1,23 +1,37 @@
 # frozen_string_literal: true
 
+require "rqrcode"
+
 module Decidim
   class InvalidUrlError < StandardError; end
 
   class LinksController < Decidim::ApplicationController
     skip_before_action :store_current_location
 
+    include Decidim::OrganizationHelper
+    include Decidim::QrCodeHelper
     helper Decidim::ExternalDomainHelper
-    helper_method :external_url
+    helper_method :external_url, :resource, :qr_code, :qr_code_image
+
+    before_action :parse_url, only: :new
 
-    before_action :parse_url
     rescue_from Decidim::InvalidUrlError, with: :modal
     rescue_from URI::InvalidURIError, with: :modal
 
+    layout false, only: :qr
+
     def new
       headers["X-Robots-Tag"] = "none"
       headers["Link"] = %(<#{url_for}>; rel="canonical")
     end
 
+    def qr
+      respond_to do |format|
+        format.html
+        format.png { send_data(qr_code.as_png(size: 480), filename: "qr-#{organization_name}.png") }
```

Comment:
Instead of using the organization name (that's unique, so if you download more than one QR code then you will not know what is each of these), this file should be the sluggified title of the resource (or something like `qr-#{organization_name}-#{sluggified_title}.png`)

Reply:
added as part of 6d7c679a02